packages/material-ui/src/Container/Container.d.ts

```diff
@@ -3,8 +3,23 @@ import { OverridableComponent, OverrideProps } from '../OverridableComponent';
 
 export interface ContainerTypeMap<P = {}, D extends React.ElementType = 'div'> {
   props: P & {
+    children: NonNullable<React.ReactNode>;
```

Comment:
Unfortunately, this is a breaking change. `null` and `undefined` were previously allowed as `children` but updating to v4.10.2 breaks code that relies on implicit `React.ReactNode` type.

Reply:
These always logged runtime, dev-only warnings. Though we're considering revisiting required children for v5.

Reply:
Aha, OK, I've never noticed that warning… Not sure though that the restriction makes sense; it should disallow `false` as well and even then it wouldn't prevent rendering `Container` without content as an empty string, which is effectively the same, could still be passed.

Reply:
Sure, but just because a solution isn't perfect doesn't mean we shouldn't try to. The idea was to report the most common mistakes. You can always silence TS by passing `<Container>{children!}</Container>`.